### ISSUE TITLE
Automate accepting PRs of signatories

### ISSUE THREAD

**Issue by KOLANICH** [rms-support-letter/rms-support-letter.github.io#2553]
It is completely possible to automate signing. Most of signatorees are github users and put their GH page as a link. It is completely possible to automatically verify that a signatoree and its commit matches the following conditions:

0. the PR only adds a YAML file, that matches a certain JSONSchema
1. Proof of "ownership":
    * the link is to the GH account that has sent the PR

2. the account is a real user.
    * A real user is the one that has at least one of:
       * explicitly marked as such
       * has accepted contributions to real open-source projects.
    * A real project is the one that has at least one of:
        * explicitly marked as such
        * has all of:
            * which owner is a real user
            * has at least 5 stars of real users except its owner
            * has at least 2 contributions of real users except its owners

A GitHub action can be created that wakes once in 30 minutes and automatically msrges all the PRs it detects as real.

Also the same can be done with a special issue. No need to fork, just post a message into a special issue with a text having a certain template, and the bot will do the checks and add a signature itself.

**Comment by nukeop** [comment]
There are still people trying to smuggle in all sorts of crap in the PRs, and I think manual merging is necessary to verify authenticity. There is stuff that simple automation will miss, and we'll end up with an embarrassing number of signatures from Sadie Enward and I.C. Wiener. We can process signatures pretty fast now, it's not a huge problem.

**Comment by d-sfounis** [comment]
@nukeop can you tell us some examples of people trying to degrade the support letter?  
Might be useful later on

**Comment by cmd410** [comment]
> some examples of people trying to degrade the support letter?

I definitely saw someone trying to inject javascript in their sign, it wouldn't work of course, but still not cool

**Comment by nukeop** [comment]
> @nukeop can you tell us some examples of people trying to degrade the support letter?
> Might be useful later on

#875

example why I still need to review everything manually

**Comment by nukeop** [comment]
Looks like it was deleted altogether (probably by Github staff?)

**Comment by gorshunovr** [comment]
There is a number of correctly created PRs which were closed by Authors: https://github.com/rms-support-letter/rms-support-letter.github.io/pulls?q=is%3Apr+is%3Aclosed+is%3Aunmerged. Is there a problem with wording in instructions which leads people to close PR before merge?
Might be a good idea to go over a few of those unmerged PRs and ask what has happened.

**Comment by DOORAK** [comment]
You may relook closed pulls either. Some peoples was not listed just by taking a bit negative experience about no real reasons closed their few PR attempts.

**Comment by imachug** [comment]
I checked the closed PRs yesterday, and the authors of most of them have somehow committed their signatures afterwards, either via another PR or via a comment on #837, so I don't think it is a big issue. I noticed only a few users who tried to sign but failed, I messaged them directly.

As it may be harmful for us to accept signatures automatically, even for GitHub accounts (see e.g. RealGitHubAdmin's PR), and we merge the PRs quickly enough, I'm closing this issue. Thanks for your contribution!
